pareto/utilities/visualize.py

```diff
@@ -0,0 +1,208 @@
+    for i in m.s_LLA:
+        piping_arcs.append(i)
+
+    trucking_arcs = []
+    for i in m.s_LLT:
+        trucking_arcs.append(i)
+
+    if show_piping:
+        G.add_edges_from(piping_arcs)
+    if show_trucking:
+        G.add_edges_from(trucking_arcs)
+
+    if len(pos) == 0:
+        pos = nx.circular_layout(G)
+
+    plt.figure()
+
+    for node in m.s_PP:
+        nx.draw_networkx_nodes(
+            G,
+            pos,
+            [node],
+            node_shape="o",
+            node_size=500,
+            node_color="lightgreen",
+            edgecolors="black",
+        )
+
+    for node in m.s_CP:
+        nx.draw_networkx_nodes(
+            G,
+            pos,
+            [node],
+            node_shape="o",
+            node_size=500,
+            node_color="lightgreen",
+            edgecolors="black",
+        )
+
+    for node in m.s_F:
+        nx.draw_networkx_nodes(
+            G,
+            pos,
+            [node],
+            node_shape="P",
+            node_size=600,
+            node_color="lightblue",
+            edgecolors="black",
+        )
+
+    for node in m.s_K:
+        nx.draw_networkx_nodes(
+            G,
+            pos,
+            [node],
+            node_shape="X",
+            node_size=600,
+            node_color="lightgreen",
+            edgecolors="black",
+        )
+
+    for node in m.s_S:
+        nx.draw_networkx_nodes(
+            G,
+            pos,
+            [node],
+            node_shape="H",
+            node_size=400,
+            node_color="lightgreen",
+            edgecolors="black",
+        )
+
+    for node in m.s_R:
+        if m.p_chi_DesalinationSites[node]:
+            shape = "s"
+            size = 300
+        else:
+            shape = "p"
+            size = 400
+        nx.draw_networkx_nodes(
+            G,
+            pos,
+            [node],
+            node_shape=shape,
+            node_size=size,
+            node_color="lightgreen",
+            edgecolors="black",
+        )
+
+    for node in m.s_O:
+        nx.draw_networkx_nodes(
+            G,
+            pos,
+            [node],
+            node_shape="*",
+            node_size=1200,
+            node_color="lightgreen",
+            edgecolors="black",
+        )
+
+    for node in m.s_N:
+        nx.draw_networkx_nodes(
+            G,
+            pos,
+            [node],
+            node_shape=",",
+            node_size=0,
+        )
+
+    if show_piping:
+        draw_arcs(G, pos, piping_arcs)
+
+    if show_trucking:
+        draw_arcs(G, pos, trucking_arcs, color="red", width=1.5, arrowsize=10)
+
+    if show_results != None:
+        built_pipes = []
+        built_trucking = []
+        for t in m.s_T:
+            for i in piping_arcs:
+                if m.v_C_Piped[i, t].value:
+                    built_pipes.append(i)
+            for i in trucking_arcs:
+                if m.v_C_Trucked[i, t].value:
+                    built_trucking.append(i)
```

Comment:
This code seems to not have any effect on the graph which is produced. I think perhaps we should punt for now on the idea of a "show_results" feature and possibly return to it in the future.

Reply:
Can we not just call draw_arcs function in this?